libr/bin/p/bin_hunk.c

```diff
@@ -84,17 +61,31 @@ static RList* sections(RBinFile *bf) {
 	return ret;
 }
 
-static RList* entries(RBinFile *bf) { //Should be 3 offsets pointed by NMI, RESET, IRQ after mapping && default = 1st CHR
+static RList* entries(RBinFile *bf) {
 	RList *ret;
 	RBinAddr *ptr = NULL;
+	RBuffer *buf;
+	int addr;
+	ut8 b[4];
 	if (!(ret = r_list_new ())) {
 		return NULL;
 	}
 	if (!(ptr = R_NEW0 (RBinAddr))) {
 		return ret;
 	}
-	ptr->paddr = 0x24;
-	ptr->vaddr = 0x24;
+	buf = bf->buf;
+	for (addr = 0x18; addr <= (r_buf_size (buf) - 4); addr += 0x4) {
```

Comment:
Define a const variable holding this bufsize-4 before entering the loop, so thats not called. On every interation

Reply:
Also no need for hex 0x4. Just use 4. Its more readable and consistent with the rest of the code

Reply:
i could also "fix" this line
https://github.com/radareorg/radare2/blob/master/libr/bin/p/bin_pyc.c#L29
while at it, where i originally copied that hex style from

Reply:
i did commit code that precalced the `bufsize - 4`,

but in current code, where we have a bigger buffer now, that we read before the loop,
we need to keep track of position for `memcmp()` and `addr` and so we end up having a pointer addition in that very for loop anyway. only a slightly different one. now its a `b + 4` for `memcmp()`. it could be coded in different ways, but one addition / offset remains.